Upgrade to MUI v7 and refactor navbar layout

NahomAlemu opened the pull request:
- **Resolves:** #646 
- Upgrade the project to Material UI v7 to ensure compatibility with the latest MUI features and eliminate deprecated patterns such as Grid2.
- **Changes:**
  - Replaced deprecated Grid2 with Grid, and updated deprecated Grid item usage based on [MUI v7 migration codemods.](https://mui.com/material-ui/migration/upgrade-to-v7/)
  - Replaced Grid2 usage with Grid and updated Grid props for MUI v7.
  - Ran the following official codemods for safe and consistent upgrades:
    `npx @mui/codemod v7.0.0/grid-props ./component`
    `npx @mui/codemod v7.0.0/grid-props ./app`
  -  **Navbar Layout Refactor**
  -  Fixed header layout issue after upgrade by using Box instead of nested Grid item. (Buttons in header (Events, Sign In, Theme Toggle) were left-aligned due to Grid layout shift.)
  - If preferred, this can be split into a follow-up PR.

## Screenshots

<details>
  <summary>Expand ⬇️</summary>

  ![navbar](https://github.com/user-attachments/assets/83d698d9-3f86-48cf-a135-a26edb9c75af)

</details>

## Expected Behavior:
- UI components function as before, with no layout shifts or errors introduced from deprecated usage. All Grid usage complies with MUI v7 standards.
- No Grid2 usage
- DatePicker works as expected
- No layout or styling breaks
## 💡 Note:
Prettier applied formatting changes automatically. We could consider standardizing it across the team for consistency?

Alzavio commented:
All the pages look normal and nothing broke. Looks good.

Alzavio commented:
Looks great overall. I did notice this issue though that's not related to this PR when checking all the pages. We may want to log a bug fix. 
![Screenshot 2025-04-20 at 20-48-10 NSC Events](https://github.com/user-attachments/assets/e8c1a8db-652d-4a7f-bd9a-6dbe74a87f38)

NahomAlemu commented:
> Looks great overall. I did notice this issue though that's not related to this PR when checking all the pages. We may want to log a bug fix.

Thanks for pointing that out. I’d be happy to take it on. I’ll track it as a separate issue and follow up with a fix.

Asfand00 reviewed (changes requested):
Nice job @NahomAlemu overall, very hefty work upgrading to MUI v7. I like that you used that guide with the codemods to upgrade everything safe and consistent. I'm not noticing any issues on the pages, and everything seems to build and compile correctly. 

![image](https://github.com/user-attachments/assets/210713f3-1358-48f5-b2b8-3dc14108cf4e)
I am wondering what your screenshot is that you had under expand, maybe you can explain that @NahomAlemu  because I don't see that layout on my end for the navbar. 

Other than that, the only issue is that we need to remove the karats "^" from your package.json as we want to enforce stability within our projects so were not always upgrading dependencies out of the blue to the latest. We want to control updating our dependencies.

NahomAlemu commented:
>  the only issue is that we need to remove the karats "^" from your package.json as we want to enforce stability within our projects so were not always upgrading dependencies out of the blue to the latest. We want to control updating our dependencies.

Sounds good, thank you for that. Pushed a new commit removing carets (^) from MUI-related dependencies. 

During this update, the build failed because there is currently no stable version of @mui/lab compatible with @mui/material v7.0.2. The previous @mui/lab@7.0.0-beta.2 worked due to the flexibility allowed by the caret (^).

After reviewing the codebase, I confirmed that @mui/lab is not actively used anywhere. Therefore, I removed it from package.json to avoid unnecessary dependency conflicts. If needed in the future, we can reinstall it once a stable, compatible version becomes available.

> I am wondering what your screenshot is that you had under expand, maybe you can explain

Yes, that was related to the navbar update. I had noted it in my PR description.
To give more context: after upgrading to MUI v7, the header layout broke — just like the screenshot.
This was due to changes in Grid behavior and the auto removal of Grid's item usage.

To resolve this cleanly, I refactored the layout by replacing the nested Grid structure with Box components using display: "flex" and justifyContent: "space-between".

Since the header layout was relatively simple (logo on the left, buttons on the right), I thought using Box offered a better, cleaner, and more maintainable solution post-upgrade.

gitbiruk2010 reviewed (approved):
Nice work on the mui v7 migration. all layouts render correctly with no visual regressions and all responsive behavior matches. looks great on my end. 

https://github.com/user-attachments/assets/1da374c5-e0d3-49d0-9642-0d4d8d1129a4

Bejarano03 commented:
LGTM, great work on getting these migrations working. I was able to recreate this on my local machine and the UI components are able to load up. I too think that you did a great job in implementing the Box component as it would work better for this situation. Good job throughout

bcko reviewed (approved):
Great job. Let's get this merged and give students a quick overview next week in class the difference between the v7 and the previous version.

Asfand00 reviewed (approved):
Looks good overall @NahomAlemu, I'm currently trying to resolve a bug concerning the homeeventcard component and its overlapping issue of the event cards. discovered that it may actually be the homepage instead and the way Grid is being used on the page so maybe this PR might help fix that issue because its currently having overload errors with how the props are being recognized by either Grid2 or Grid so hopefully this will help. 

Your package.json looks fine for the most part, sorry i meant to say to remove every karat from every dependecy not just MUI, so there was one karat (^) that I saw but, if need be, I can go and remove it myself in a small PR so no worries for now, 
![image](https://github.com/user-attachments/assets/320598ea-f0ac-440c-852e-34150e9fd1ca)

Thanks for your work and your explanation/clarification about the MUI v7 stuff and what's going on there (your reply post to me).
